modules/openapi-generator/src/main/resources/typescript-redux-query/apis.mustache

```diff
@@ -0,0 +1,248 @@
+// tslint:disable
+{{>licenseInfo}}
+
+import { HttpMethods, QueryConfig, ResponseBody, ResponseText } from 'redux-query';
+import * as runtime from '../runtime';
+
+{{#imports.0}}
+import {
+    {{#imports}}
+    {{className}},
+    {{className}}FromJSON,
+    {{className}}ToJSON,
+    {{/imports}}
+} from '../models';
+{{/imports.0}}
+
+{{#operations}}
+{{#operation}}
+{{#allParams.0}}
+export interface {{operationIdCamelCase}}Request {
+    {{#allParams}}
+    {{paramName}}{{^required}}?{{/required}}: {{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}};
+    {{/allParams}}
+}
+
+{{/allParams.0}}
+{{/operation}}
+{{/operations}}
+{{#operations}}
+
+{{#operation}}
+/**
+ {{#notes}}
+ * {{&notes}}
+ {{/notes}}
+ {{#summary}}
+ * {{&summary}}
+ {{/summary}}
+ */
+function {{nickname}}Raw<T>({{#allParams.0}}requestParameters: {{operationIdCamelCase}}Request,{{/allParams.0}} requestConfig: runtime.TypedQueryConfig<T, {{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/returnType}}> = {}): QueryConfig<T> {
+    {{#allParams}}
+    {{#required}}
+    if (requestParameters.{{paramName}} === null || requestParameters.{{paramName}} === undefined) {
+        throw new runtime.RequiredError('{{paramName}}','Required parameter requestParameters.{{paramName}} was null or undefined when calling {{nickname}}.');
+    }
+
+    {{/required}}
+    {{/allParams}}
+    let queryParameters = null;
+
+    {{#queryParams}}
+    queryParameters = {};
+
+    {{#isListContainer}}
+    if (requestParameters.{{paramName}}) {
+        {{#isCollectionFormatMulti}}
+        queryParameters['{{baseName}}'] = requestParameters.{{paramName}};
+        {{/isCollectionFormatMulti}}
+        {{^isCollectionFormatMulti}}
+        queryParameters['{{baseName}}'] = requestParameters.{{paramName}}.join(runtime.COLLECTION_FORMATS["{{collectionFormat}}"]);
+        {{/isCollectionFormatMulti}}
+    }
+
+    {{/isListContainer}}
+    {{^isListContainer}}
+    if (requestParameters.{{paramName}} !== undefined) {
+        {{#isDateTime}}
+        queryParameters['{{baseName}}'] = (requestParameters.{{paramName}} as any).toISOString();
+        {{/isDateTime}}
+        {{^isDateTime}}
+        {{#isDate}}
+        queryParameters['{{baseName}}'] = (requestParameters.{{paramName}} as any).toISOString().substr(0,10);
+        {{/isDate}}
+        {{^isDate}}
+        queryParameters['{{baseName}}'] = requestParameters.{{paramName}};
+        {{/isDate}}
+        {{/isDateTime}}
+    }
+
+    {{/isListContainer}}
+    {{/queryParams}}
+    const headerParameters = {};
+
+    {{#bodyParam}}
+    {{^consumes}}
+    headerParameters['Content-Type'] = 'application/json';
+
+    {{/consumes}}
+    {{#consumes.0}}
+    headerParameters['Content-Type'] = '{{{mediaType}}}';
```

Comment:
support for multiple media types could be added, see https://github.com/OpenAPITools/openapi-generator/pull/3849

Reply:
Do you mean around providing a specific `Accept` header here? It seems like this code producing the `Content-Type` for the request body generated is working as expected, and matches the code in #3849.

Reply:
ah, yes, I was talking about the `Accept` header.